src/sys/windows/udp.rs

```diff
@@ -13,7 +13,7 @@ use std::{fmt, io};
 use winapi::um::winsock2::{bind, closesocket, SOCKET_ERROR, SOCK_DGRAM};
 
 pub struct UdpSocket {
-    internal: Arc<Mutex<Option<InternalState>>>,
+    internal: Mutex<Option<InternalState>>,
```

Comment:
What is the size of the struct now?

Longer term, I think we can optimize `InternalState` to be a single arc, but that can happen later.

Reply:
Currently it's 56 bytes (https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=2058e09e5c44b46dca119ca7165dec16). In the future it might change to use parking lot (https://github.com/rust-lang/rust/pull/56410) in which it's a single bit/byte.

Want to change to a `Box`? That brings it down to 16 (on the stack).

Reply:
@carllerche what do you want to here? I don't know how much the net types are moved around normally.